### PR TITLE
Add DisabilityGroup type, support populating assessment from it

### PR DESCRIPTION
**Pull request by gigxz**
`disabilityGroups` resolves a list of unpaginated (max 25) objects that each represent a "group" of Disability records (records that share Information Date, Data Collection Stage, User, and Enrollment).

### CODE REVIEW

**Comment by gigxz** [comment]
UI:

Screen Shot 2022-11-29 at 4 40 06 PM

File: drivers/hmis/app/graphql/types/hmis_schema/disability_group.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Types
+  class HmisSchema::DisabilityGroup < Types::BaseObject
+    description 'Group of disability records that were collected at the same time'
+
+    field :id, ID, null: false
```
Comment (eanders) [inline drivers/hmis/app/graphql/types/hmis_schema/disability_group.rb]: which `id` is this?  Won't each disability have different id?

File: drivers/hmis/app/graphql/types/hmis_schema/has_disability_groups.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+# frozen_string_literal: true
+
+module Types
+  module HmisSchema
+    module HasDisabilityGroups
+      extend ActiveSupport::Concern
+      include ArelHelper
+
+      class_methods do
+        def disability_groups_field(name = :events, description = nil, **override_options, &block)
```
Comment (eanders) [inline drivers/hmis/app/graphql/types/hmis_schema/has_disability_groups.rb]: Why does name default to `:events`?

File: drivers/hmis/app/graphql/types/hmis_schema/has_disability_groups.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        result_aggregations = result_fields.map { |f| array_agg(d_t[f]).to_sql }
+
+        disability_groups = scope.viewable_by(current_user).
+          order(information_date: :desc, data_collection_stage: :desc).
+          limit(150). # No pagination, so LIMIT to 25 most recent groups (6*25 = 150 records)
+          group(*key_fields).
+          pluck(*key_fields, *result_aggregations)
+
+        enrollment_ids = disability_groups.map(&:first).uniq
+        user_ids = disability_groups.map(&:second).uniq
```
Comment (eanders) [inline drivers/hmis/app/graphql/types/hmis_schema/has_disability_groups.rb:52]: This seems somewhat dangerous.  I wonder if it's worth fetching the location of the field from the array.  As is, if the array order changes, or a new item is added to the beginning, these will break.  We should add a comment above alongside the array that the order matters and is used further down, so don't change them.

**Review by eanders — APPROVED** [review]
Looks good to me.  Left a few comments inline.

File: drivers/hmis/app/graphql/types/hmis_schema/disability_group.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/types/hmis_schema/disability_group.rb]: It's a concatenated list of all six ids (`1:2:3:4:5:6`). We need an ID for apollo caching, and its helpful for testing too to validate that the correct records are grouped together. I'll add a comment

File: drivers/hmis/app/graphql/types/hmis_schema/has_disability_groups.rb (hunk 2 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/types/hmis_schema/has_disability_groups.rb]: Copy-paste error, thanks

File: drivers/hmis/app/graphql/types/hmis_schema/disability_group.rb (hunk 1 above)
Comment (eanders) [reply drivers/hmis/app/graphql/types/hmis_schema/disability_group.rb]: Sounds good.  Thanks!

File: drivers/hmis/app/graphql/types/hmis_schema/has_disability_groups.rb (hunk 3 above)
Comment (gigxz) [reply drivers/hmis/app/graphql/types/hmis_schema/has_disability_groups.rb:52]: Added comments

**Pull request merged**